### ISSUE TITLE
Stream opening should be lazy

### ISSUE THREAD

**Issue by Stebalien** [libp2p/go-mplex#30]
Currently, we end up sending a bunch of tiny open packets.

**Comment by raulk** [comment]
@Stebalien wondering if we have an analysis of the packets we are sending, and their contents? 
Also, do you think there are two issues here, i.e. opening streams lazily _and_ reducing opening traffic?

**Comment by Stebalien** [comment]
Right. Much of that will be improved with https://github.com/libp2p/go-mplex/pull/29. However, we'll *still* end up sending a separate opening packet followed by the message.

So, I guess at this point, bunch = 2.
